packages/devtools_app/lib/src/shared/editable_list.dart

```diff
@@ -0,0 +1,272 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import 'common_widgets.dart';
+import 'theme.dart';
+import 'utils.dart';
+
+class EditableList extends StatefulWidget {
+  const EditableList({
+    required this.entries,
+    required this.textFieldLabel,
+    this.isRefreshing,
+    this.onEntryAdded,
+    this.onEntryRemoved,
+    this.onRefresh,
+  });
+
+  final ValueListenable<List<String>> entries;
+  final ValueListenable<bool>? isRefreshing;
+  final Function(String)? onEntryAdded;
+  final Function(String)? onEntryRemoved;
+  final Function()? onRefresh;
+  final String textFieldLabel;
+
+  @override
+  State<StatefulWidget> createState() => _EditableListState();
+}
+
+class _EditableListState extends State<EditableList> {
+  @override
+  void initState() {
+    super.initState();
+    textFieldController = TextEditingController();
+  }
+
+  late final TextEditingController textFieldController;
+  final FocusNode textFieldFocusNode = FocusNode();
+
+  @override
+  void dispose() {
+    textFieldController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return DualValueListenableBuilder(
+      firstListenable: widget.entries,
+      secondListenable: widget.isRefreshing ?? ValueNotifier<bool>(false),
+      builder: (_, __, ___, ____) {
+        return Column(
+          mainAxisSize: MainAxisSize.min,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            EditableListActionBar(
+              textFieldFocusNode: textFieldFocusNode,
+              textFieldController: textFieldController,
+              isRefreshing: widget.isRefreshing,
+              textFieldLabel: widget.textFieldLabel,
+              onEntryAdded: widget.onEntryAdded,
+              onRefresh: widget.onRefresh,
+            ),
+            Flexible(
```

Comment:
nit: use Expanded instead here - this makes it clear that the intent is for this widget to take the remaining space in the column